test/model/content_test.dart

```diff
@@ -521,6 +521,24 @@ class ContentExample {
       blockUnimplemented('more text'),
     ]]),
   ]);
+
+  static const thematicBreaks =  ContentExample(
+    'parse thematic break (<hr>) in block context',
+    '---'
+    '---'
+    'a'
+    '---'
+    'b'
+    '---',
+    '<hr><hr><p>a</p><hr><p>b</p><hr>',
```

Comment:
Then as a style nit: prefer to put these strings on relatively few lines here in the test source code, writing `\n` where necessary inside the string literals. When there's a lot going on in the string, it can use more lines in order to keep it readable; but this one is fairly simple, so `markdown` and `html` should use one line of source code each.

Reply:
A higher-level point of testing style: generally we try to keep each test case tightly focused, containing only the details that are needed in order to make the point of the test.

This test case feels like it's gesturing at several different things; it can be improved by being made shorter, to focus on just the minimum for demonstrating an `<hr>` element. Those other things (like two `<hr>` in a row?) can be exercised with their own focused tests, though I'm not sure we need tests for them.

The absolute minimum would be just `---`. Because it'd be kind of odd to send a message in normal use that's just a bare `<hr>`, it'd be reasonable to make it slightly longer to resemble normal use: like `a\n---\nb`. But then that's plenty.